### PR TITLE
issue #309: custom stepper implementation

### PR DESCRIPTION
**Pull request by k-allagbe**
- [x] improvements to the theme
- [x] custom stepper implementation
- [x] tests

### CODE REVIEW

**Comment by SamuelPelletierEvraire** [comment]
Shouldn't we use all the same way of style ?

**Comment by k-allagbe** [comment]
> Shouldn't we use all the same way of style for better conformity ?

Do you mean css? Yes. But then I would enforce Tailwind because it works well with mui and non mui components and integrates well with traditional .css files. Give this extension a try: [Tailwind CSS IntelliSense](https://marketplace.visualstudio.com/items?itemName=bradlc.vscode-tailwindcss).

**Comment by SamuelPelletierEvraire** [comment]
Ok perfect I'm gonna try to translate SideNav into tailwind.

File: src/components/HorizontalNonLinearStepper.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  steps,
+  activeStep,
+  stepStatuses,
+  setActiveStep,
+}: CustomStepperProps) {
+  const handleStep = (step: number) => () => setActiveStep(step);
+
+  return (
+    <Box sx={{ width: "100%" }}>
+      <Stepper nonLinear activeStep={activeStep}>
```
Comment (ChromaticPanic) [inline src/components/HorizontalNonLinearStepper.tsx:34]: use tailwind class ?

File: src/app/label-data-validation/page.tsx [hunk 2]
```diff
@@ -1,29 +1,153 @@
...
+  return (
+    <>
+      <Typography sx={{ mt: 2, mb: 1, py: 1 }}>
+        {allStepsCompleted
+          ? "All steps completed - you're finished"
+          : `Step ${activeStep + 1}`}
+      </Typography>
+
+      <Box sx={{ display: "flex", justifyContent: "center", gap: 2, pt: 2 }}>
+        <Button
```
Comment (ChromaticPanic) [inline src/app/label-data-validation/page.tsx:110]: Tailwind classes?

**Review by ChromaticPanic — CHANGES_REQUESTED** [review]

File: src/components/HorizontalNonLinearStepper.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  >;
+}
+
+export default function HorizontalNonLinearStepper({
+  steps,
+  activeStep,
+  stepStatuses,
+  setActiveStep,
+}: CustomStepperProps) {
+  const handleStep = (step: number) => () => setActiveStep(step);
```
[the remark this answers is not in the document]
Comment (k-allagbe) [reply src/components/HorizontalNonLinearStepper.tsx:30]: Yes I reuse it. Which is why I named it with more generically.

File: src/components/HorizontalNonLinearStepper.tsx (hunk 1 above)
Comment (k-allagbe) [reply src/components/HorizontalNonLinearStepper.tsx]: Done.

File: src/app/label-data-validation/page.tsx (hunk 2 above)
Comment (k-allagbe) [reply src/app/label-data-validation/page.tsx]: Done. Will update other components not part of this PR as well, progressively.

**Pull request merged**